components/Switch.tsx

```diff
@@ -0,0 +1,401 @@
+import { useForm, UseFormReturn } from "react-hook-form";
+import { z } from "zod";
+import { AccountInfo, ErrorInfo, FormDataSwitch } from "@/utils/types";
+import { SelectedPolkadotAccount } from "./SelectedPolkadotAccount";
+import { SelectAccount } from "./SelectAccount";
+import { Input } from "./ui/input";
+import { Button } from "./ui/button";
+import { BusyDialog } from "./BusyDialog";
+import { SendErrorDialog } from "./SendErrorDialog";
+import { SubmittableExtrinsic } from "@polkadot/api/types";
+import { ISubmittableResult } from "@polkadot/types/types";
+import PolkadotBalance from "./Balances";
+import { parseUnits } from "ethers";
+import { toast } from "sonner";
+import { parachainConfigs } from "@/utils/parachainConfigs";
+
+import { useRouter } from "next/navigation";
+import { TopUpXcmFee } from "./TopUpXcmFee";
+import { toPolkadot } from "@snowbridge/api";
+import { LocationSelector } from "./LocationSelector";
+
+export const SwitchComponent: FC = () => {
+  const snowbridgeEnvironment = useAtomValue(snowbridgeEnvironmentAtom);
+  const context = useAtomValue(snowbridgeContextAtom);
+  const polkadotAccounts = useAtomValue(polkadotAccountsAtom);
+  const polkadotAccount = useAtomValue(polkadotAccountAtom);
+  const router = useRouter();
+
+  const [feeDisplay, setFeeDisplay] = useState("");
+  const [error, setError] = useState<ErrorInfo | null>(null);
+  const [busyMessage, setBusyMessage] = useState("");
+  const [sufficientTokenAvailable, setSufficientTokenAvailable] =
+    useState(true);
+  const [topUpCheck, setTopUpCheck] = useState(false);
+  const [transaction, setTransaction] = useState<SubmittableExtrinsic<
+    "promise",
+    ISubmittableResult
+  > | null>(null);
+  const [tokenSymbol, setTokenSymbol] = useState<string | null>(null);
+
+  const filteredLocations = useMemo(
+    () =>
+      snowbridgeEnvironment.locations
+        .filter((x) => x.type !== "ethereum")
+        .filter((x) => x.name !== "Muse"),
+    [snowbridgeEnvironment],
+  );
+
+  const initialDestination = useMemo(
+    () => filteredLocations.find((v) => v.id === "KILT"),
+    [filteredLocations],
+  );
+
+  const form: UseFormReturn<FormDataSwitch> = useForm<
+    z.infer<typeof formSchemaSwitch>
+  >({
+    resolver: zodResolver(formSchemaSwitch),
+    defaultValues: {
+      source: filteredLocations.find((v) => v.id === "assethub"),
+      // to do: handle this better
+      destination: initialDestination ?? "",
+      token: initialDestination?.erc20tokensReceivable[0].id,
+      beneficiary: polkadotAccount?.address ?? "",
+      sourceAccount: polkadotAccount?.address ?? "",
+      amount: "0.0",
+    },
+  });
+
+  const source = form.watch("source");
+  const destination = form.watch("destination");
+  const sourceAccount = form.watch("sourceAccount");
+  const beneficiary = form.watch("beneficiary");
+  const amount = form.watch("amount");
+  const token = form.watch("token");
```

Comment:
One more remark: make sure that all handlers of text input fields in this form are 'debounced'! (see https://softwaremill.com/debounce-on-inputs-in-react/ for an explanation)
If not, every letter or number entered, for example, into the amount field will immediately trigger extrinsic generation and fee estimation, which will result in network requests to the full node. This can lead to race conditions and may get you blocked/throttled by the node.

Reply:
https://github.com/Snowfork/snowbridge-app/pull/23/commits/065e0b6f072ae31ffdb94f89e2688d1b6ec74a87